Intel asciidoc extensions

bashbaug opened the pull request:
This pull request adds the asciidoc spec source for several Intel OpenCL extensions:

* cl_intel_subgroups
* cl_intel_subgroups_short
* cl_intel_required_subgroup_size
* cl_intel_spirv_subgroups
* cl_intel_spirv_media_block_io
* cl_intel_spirv_device_side_avc_motion_estimation

oddhack commented:
Oh, never mind my email comment asking where the sources were. BTW I think .adoc is more customary than .asciidoc.

alycm commented:
Is it intended that vendor extensions should also be added to extensions,txt?

bashbaug commented on extensions/Makefile:
```diff
@@ -19,7 +19,7 @@
 #
 
-all: html pdf
+all: html
```
Fixed in 23839bf.

bashbaug commented:
> Is it intended that vendor extensions should also be added to extensions,txt?

I initially envisioned extensions.txt as a way to create a single doc with all vendor extensions enabled, but this isn't something that's currently published or even built by default.  Is there a need for this capability? 
 If not, perhaps it should be removed, for now at least.

alycm commented:
> I initially envisioned extensions.txt as a way to create a single doc with all vendor extensions enabled, but this isn't something that's currently published or even built by default. Is there a need for this capability?
If not, perhaps it should be removed, for now at least.

I can see that having a single document with all vendor extensions could be nice, the vendor extensions added in this PR have not been added to extensions.txt though.  Also, the comments in extensions.txt talk about khr etc extensions, not vendor extensions.

However, my original question was really just intended as "should these vendor extensions be added to extensions.txt in this PR?", and in practice it seems like there is no reason to do that currently.

bashbaug commented:
I went ahead and added the Intel extensions to extensions.txt just in case we decide to resurrect this capability.  I also refactored the doc slightly so the comments are more clear what it is intended to do.

The HTML output looks pretty good with these changes.  The PDF output will need some work though (the table of contents has too many levels and some of the table text renders poorly, among other things).

I'd like this PR to be accepted soon if there are no other issues - when I uploaded the HTML output to the registry for the subgroup extensions I also uploaded a new txt document linking to the HTML output and to the asciidoc source, and the asciidoc source link will be broken until this PR is accepted.  Thanks!

alycm commented:
LGTM.
